gtk/src/toga_gtk/window.py

```diff
@@ -75,11 +76,23 @@ def gtk_window_state_event(self, widget, event):
             current_state = self.get_window_state()
             if current_state != WindowState.NORMAL:
                 if self._pending_state_transition != current_state:
-                    self._apply_state(WindowState.NORMAL)
+                    # Add slight delay to prevent glitching  on wayland during rapid
+                    # state switching.
+                    if IS_WAYLAND:  # pragma: no-cover-if-linux-x
+                        GLib.timeout_add(
+                            10, partial(self._apply_state, WindowState.NORMAL)
```

Comment:
Adding "magic numbers" is borderline behavior in a test; but in production code, it's *really* not a good idea - who is to say that 10ms is enough? It's inherently going to be machine dependent; and as machines change over time, it's unclear how performance changes will impact the chosen "magic" number.

Reply:
I agree, I will try to find any Gtk callback event that would be triggered on completion of Gtk side processing.

Reply:
I tried to use `Gtk.events_pending()` and `Gtk.main_iteration_do()`, but it still didn't work:
```diff
diff --git a/gtk/src/toga_gtk/window.py b/gtk/src/toga_gtk/window.py
index f2a0e1448..1c4407f8a 100644
--- a/gtk/src/toga_gtk/window.py
+++ b/gtk/src/toga_gtk/window.py
@@ -79,20 +79,16 @@ class Window:
                     # Add slight delay to prevent glitching  on wayland during rapid
                     # state switching.
                     if IS_WAYLAND:  # pragma: no-cover-if-linux-x
-                        GLib.timeout_add(
-                            10, partial(self._apply_state, WindowState.NORMAL)
-                        )
-                    else:  # pragma: no-cover-if-linux-wayland
-                        self._apply_state(WindowState.NORMAL)
+                        while Gtk.events_pending():
+                            Gtk.main_iteration_do(False)
+                    self._apply_state(WindowState.NORMAL)
                 else:
                     self._pending_state_transition = None
             else:
                 if IS_WAYLAND:  # pragma: no-cover-if-linux-x
-                    GLib.timeout_add(
-                        10, partial(self._apply_state, self._pending_state_transition)
-                    )
-                else:  # pragma: no-cover-if-linux-wayland
-                    self._apply_state(self._pending_state_transition)
+                    while Gtk.events_pending():
+                        Gtk.main_iteration_do(False)
+                self._apply_state(self._pending_state_transition)
 
     def gtk_delete_event(self, widget, data):
         # Return value of the GTK on_close handler indicates whether the event has been
         ```

Reply:
Also tried `Gtk.idle_add()` and it also didn't work:
```diff
diff --git a/gtk/src/toga_gtk/window.py b/gtk/src/toga_gtk/window.py
index f2a0e1448..dea71f108 100644
--- a/gtk/src/toga_gtk/window.py
+++ b/gtk/src/toga_gtk/window.py
@@ -79,17 +79,15 @@ class Window:
                     # Add slight delay to prevent glitching  on wayland during rapid
                     # state switching.
                     if IS_WAYLAND:  # pragma: no-cover-if-linux-x
-                        GLib.timeout_add(
-                            10, partial(self._apply_state, WindowState.NORMAL)
-                        )
+                        GLib.idle_add(partial(self._apply_state, WindowState.NORMAL))
                     else:  # pragma: no-cover-if-linux-wayland
                         self._apply_state(WindowState.NORMAL)
                 else:
                     self._pending_state_transition = None
             else:
                 if IS_WAYLAND:  # pragma: no-cover-if-linux-x
-                    GLib.timeout_add(
-                        10, partial(self._apply_state, self._pending_state_transition)
+                    GLib.idle_add(
+                        partial(self._apply_state, self._pending_state_transition)
                     )
                 else:  # pragma: no-cover-if-linux-wayland
                     self._apply_state(self._pending_state_transition)
                     ```

Reply:
I have also tested with the following signals, and still they do not work consistently:
* `map-event`
* `realize`
* `draw`
* `configure-event`
* `state-flags-changed`

Reply:
I tried to find any reliable alternative to the timeout method, but unfortunately, none of them work consistently and properly.

Reply:
Hrm... I don't like it, but I also don't have any better suggestions. I guess I can live with it as long as we add an inline comment about why the delay is there.